core/dnsserver/server_https.go

```diff
@@ -50,9 +50,11 @@ func NewServerHTTPS(addr string, group []*Config) (*ServerHTTPS, error) {
 	// The *tls* plugin must make sure that multiple conflicting
 	// TLS configuration returns an error: it can only be specified once.
 	var tlsConfig *tls.Config
-	for _, conf := range s.zones {
-		// Should we error if some configs *don't* have TLS?
-		tlsConfig = conf.TLSConfig
+	for _, z := range s.zones {
+		for _, conf := range z {
+			// Should we error if some configs *don't* have TLS?
+			tlsConfig = conf.TLSConfig
+		}
```

Comment:
I just noticed that this will be a problem if multiple configs for the same zone have different TLS configs. Not sure if this is easy to fix.

Reply:
Although, this already a problem for different server blocks that listen on the same port (i.e. same server).  Despite the comment above this, _tls_ does not in fact enforce one TLS definition per server, it just enforces one defined per Config (each server block creates a separate Config. each Server can serve multiple Configs).

So I don't think this introduce a new problem.  But the existing problem should be fixed - or documented.